pages/register.tsx

```diff
@@ -0,0 +1,290 @@
+                                >
+                                    {
+                                        branchNames.map((branchName, id) => <MenuItem value={branchName} key={id}>{branchName}</MenuItem>)
+                                    }
+                                </Select>
+                            </FormControl>
+
+                            <div>
+                                <TextField label="SID"
+                                    name="sid"
+                                    variant="filled"
+                                    error={!isSidCorrect(formValues.sid)}
+                                    onChange={handleChange}
+                                    value={formValues.sid}
+                                    fullWidth
+                                    required />
+                                {
+                                    isSidCorrect(formValues.sid)
+                                        ? <></>
+                                        : <ErrorTextBox text={errorText.invalidSid} icon={<BiErrorCircle />} />
+                                }
+                            </div>
+                        </div>
+                    </div>
+
+                </div>
+
+                {/* User Account Details */}
+                <div className={styles.sectionWrapper}>
+
+                    <div className={styles.flowSection}>
+                        <div className={styles.accountDetails}>
+                            <TextField
+                                name="username"
+                                label="Username"
+                                variant="filled"
+                                onChange={handleChange}
+                                value={formValues.username}
+                                fullWidth
+                                required />
+                            <div className={styles.smallText}>This will be used to login from now on.</div>
+                        </div>
+                        <div className={styles.accountDetails}>
+                            <TextField
+                                name="email"
+                                label="name@email.com"
+                                variant="filled"
+                                type="email"
+                                onChange={handleChange}
+                                value={formValues.email}
+                                fullWidth
+                                required />
+                        </div>
+                        <div className={styles.accountDetails}>
+                            <TextField
+                                name="password"
+                                label="Password"
+                                variant="filled"
+                                type="password"
+                                onChange={handleChange}
+                                error={isPasswordStrong(formValues.password) !== 2}
```

Comment:
This thing has a bug, when i visit the screen, this is already throwing error without me entering anything.
<img width="1371" alt="Screenshot 2023-07-07 at 10 59 07 PM" src="https://github.com/PEC-CSS/website/assets/50266759/abf9d573-8db2-4994-81f1-ffe7b73ab031">